spec/unit/commands/service_command/create_command_spec.rb

```diff
@@ -0,0 +1,95 @@
+ require '3scale_toolbox'
+
+RSpec.describe ThreeScaleToolbox::Commands::ServiceCommand::ServiceCreateSubcommand do
+  include_context :random_name
+
+  context '#run' do
+    let(:remote) { double('remote') }
+    let(:options) { {} }
+
+    subject { described_class.new(options, arguments, nil) }
+
+    before :example do
+      expect(subject).to receive(:threescale_client).with('myremote').and_return(remote)
+    end
+
+    # This should cover the case where the service already exists
+    # or other errors that are returned by calls to the API
+    context "when there is an error creating the service" do
```

Comment:
AS commented above the test is correct for the current implementation. I would change implementation to handle `system_name has already been taken`.

This test would still be valid for any other kind of error